src/projector_monte_carlo_problem.jl

```diff
@@ -181,6 +184,12 @@ function ProjectorMonteCarloProblem(
 
     n_replicas = num_replicas(replica_strategy) # replica_strategy may override n_replicas
 
+    n_spectral = num_spectral_states(spectral_strategy) # spectral_strategy may override n_spectral
+
+    if replica_strategy == AllOverlaps(n_replicas) && n_spectral > 1
```

Comment:
```suggestion
    if replica_strategy isa AllOverlaps && n_spectral > 1
```
What you wrote also works, but this is a bit more concise and simpler.